packages/database/src/prisma-api.ts

```diff
@@ -401,6 +402,7 @@ export function getSdk(
         } else {
           coupon = ctx.prisma.coupon.create({
             data: {
+              id: bulkCouponId,
```

Comment:
Here is the fix, this `id` value needs to be set to what we determined above, otherwise it just gets an auto-generated value that doesn't match what we put in `Purchase.bulkCouponId`.